synapse/static/client/login/js/login.js

```diff
@@ -46,20 +49,37 @@ var setFeedbackString = function(text) {
 };
 
 var show_login = function() {
-    $("#loading").hide();
-
     var this_page = window.location.origin + window.location.pathname;
     $("#sso_redirect_url").val(this_page);
 
-    if (matrixLogin.serverAcceptsPassword) {
-        $("#password_flow").show();
+    if (matrixLogin.serverAcceptsCas) {
```

Comment:
we can probably drop the cas stuff btw. It's superceded by SSO.

Reply:
Done in https://github.com/matrix-org/synapse/pull/7152/commits/45409c2ef3aa42a08ab18246038c735a40b5d394